locksmith/src/controllers/hookController.ts

```diff
@@ -15,178 +26,148 @@ const Hub = z.object({
   secret: z.string().optional(),
 })
 
-const EXPIRATION_SECONDS_LIMIT = 86400 * 90
-
-export function getExpiration(leaseSeconds?: number) {
-  const limit = leaseSeconds ?? 864000
-  if (limit > EXPIRATION_SECONDS_LIMIT) {
-    throw new Error("Lease seconds can't be greater than 90 days")
+interface HookControllerOptions {
+  leaseSeconds: {
+    limit: number
+    default: number
   }
-  return new Date(Date.now() + limit * 1000)
 }
+export class HookController {
+  // eslint-disable-next-line
+  constructor(public options: HookControllerOptions) {}
 
-export async function subscribe(
-  hub: z.infer<typeof Hub>,
-  params: Record<string, string>
-) {
-  try {
-    const hook = await Hook.findOne({
-      where: {
-        topic: hub.topic,
-        callback: hub.callback,
-      },
-    })
-
-    if (!hook) {
-      await validHubIntent(hub)
-
-      const expiration = getExpiration(hub.lease_seconds)
-
-      const createdHook = await Hook.create({
-        expiration,
-        topic: hub.topic,
-        callback: hub.callback,
-        mode: hub.mode,
-        secret: hub.secret,
-        network: params.network,
-        lock: params.lock,
-      })
-
-      return createdHook
+  async handle(request: SubscribeRequest, response: Response) {
+    const network = this.getNetwork(request.params.network)
```

Comment:
I wonder if this should eventually live in a middleware because we have a lot of routes that are network specific...

Reply:
(for later!)

Reply:
Make sense.